File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
import java.util.*;
class Solution {
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        
        int[] dr = {-1,1,0,0};
        int[] dc = {0,0,-1,1};
        
        for(int i=0; i<4; i++) {
            try {
                if(board[h+dr[i]][w+dc[i]].equals(board[h][w])) answer++;
            } catch(Exception e) {
                continue;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    answer = 0
    
    dr = [-1, 0, 1, 0]
    dc = [0, 1, 0, -1]
    
    for i in range(4) :
        r = dr[i] + h
        c = dc[i] + w
        
        if r<0 or c<0 or r>=len(board) or c>=len(board[0]) :
            continue
        
        if board[r][c] == board[h][w] :
            answer += 1
    
    return answer